service/server.py

```diff
@@ -197,68 +175,57 @@ def find_titles(search_term=''):
         return render_template('search.html', asset_path='/static/',
                                google_api_key=GOOGLE_ANALYTICS_API_KEY, form=TitleSearchForm())
     # search for something
-    LOGGER.info(
-        "SEARCH REGISTER: '{0}' was searched by {1}".format(
-            search_term,
-            current_user.get_id()))
+    LOGGER.info("SEARCH REGISTER: '{0}' was searched by '{1}'".format(search_term,
+                                                                      current_user.get_id()))
     # Determine search term type and preform search
     title_number_regex = re.compile(TITLE_NUMBER_REGEX)
     postcode_regex = re.compile(address_utils.BASIC_POSTCODE_REGEX)
     search_term = search_term.upper()
     # If it matches the title number regex...
     if title_number_regex.match(search_term):
-        title = get_register_title(search_term)
+        title_ref = search_term
+        title = get_register_title(title_ref)
         if title:
             # If the title exists store it in the session
             session['title'] = title
-            # Redirect to the display_title method to display the digital
-            # register
-            return redirect(url_for('display_title', title_ref=search_term))
+            # Redirect to the display_title method to display the digital register
+            return redirect(url_for('display_title', title_ref=title_ref))
         else:
             # If title not found display 'no title found' screen
-            return render_search_results([], search_term)
+            return render_search_results([], search_term, page_num)
     # If it matches the postcode regex ...
     elif postcode_regex.match(search_term):
         # Short term fix to enable user to search with postcode without spaces
         postcode = sanitise_postcode(search_term)
-        postcode_search_results = get_register_titles_via_postcode(postcode)
-        return render_search_results(postcode_search_results, postcode)
-    elif search_term:
-        address_search_results = get_register_titles_via_address(search_term)
-        return render_search_results(address_search_results, search_term)
-
+        postcode_search_results = get_register_titles_via_postcode(postcode, page_num)
+        return render_search_results(postcode_search_results, postcode, page_num)
+    else:
```

Comment:
Recently Camille made a change to this method, that would prevent search by address when the search term is empty. Your version doesn't do it anymore. Is it deliberate?